core/coreapi/block.go

```diff
@@ -48,6 +48,8 @@ func (api *BlockAPI) Put(ctx context.Context, src io.Reader, opts ...caopts.Bloc
 		return nil, err
 	}
 
+	api.node.Provider.Provide(b.Cid())
```

Comment:
You shouldn't use `api.node` outside `api.WithOption` to make sure that global options are applied correctly. `Provider` should be added to the api struct in https://github.com/ipfs/go-ipfs/blob/master/core/coreapi/coreapi.go#L47 and https://github.com/ipfs/go-ipfs/blob/master/core/coreapi/coreapi.go#L156

Reply:
This works. I had put this all in coreapi to begin with, but we opted to move it out at the time. Since this work now also covers reprovide, I'd like to move the `ipfs bitswap reprovide` command to be something more like `ipfs provider reprovide` which I think aligns with the goals of coreapi.